src/firmware/app/primitives/move_primitive.c

```diff
@@ -152,14 +103,24 @@ void app_move_primitive_start(TbotsProto_MovePrimitive prim_msg, void* void_stat
     const float current_orientation = app_firmware_robot_getOrientation(robot);
     const float current_speed       = app_firmware_robot_getSpeedLinear(robot);
 
+    const float distance_to_destination =
+        sqrtf(powf(destination_x - current_x, 2) + powf(destination_y - current_y, 2));
+    // Number of revolutions to spin, assuming the time horizon is the simplistic
+    // distance_to_destination over max_speed_m_per_s
+    const int revolutions_to_spin =
+        (int)(distance_to_destination / max_speed_m_per_s * target_spin_rps);
+    // Change in orientation to reach destination orientation
+    const float net_change_in_orientation =
+        min_angle_delta(current_orientation, destination_orientation);
+
     // Plan a trajectory to move to the target position/orientation
     FirmwareRobotPathParameters_t path_parameters = {
         .path = {.x = {.coefficients = {0, 0, destination_x - current_x, current_x}},
                  .y = {.coefficients = {0, 0, destination_y - current_y, current_y}}},
         .orientation_profile = {.coefficients = {0, 0,
-                                                 fmodf(destination_orientation -
-                                                           current_orientation,
-                                                       (float)(2 * M_PI)),
+                                                 net_change_in_orientation +
+                                                     (float)revolutions_to_spin *
+                                                         (float)(2 * M_PI),
```

Comment:
If we have to cast `(2 * M_PI)` to a `(float)` here, I suspect `2 * M_PI` is resulting in an integer, since `M_PI` is already a `float`?

Reply:
M_PI is a double :/

Reply:
whooops, my bad